### ISSUE TITLE
HMR crash when adding a new file

### ISSUE THREAD

**Issue by dmarkow** [remix-run/remix#5711]
### What version of Remix are you using?

1.14.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Created a fresh express remix app, turned on the new dev server, ran the dev server, then added a new file (in my case, I added `app/routes/about.tsx`). The dev server crashes trying to find the new route in the existing config manifest. So right now, you have to restart the dev server any time you add or move a file. Deleting files is fine.

```
> yarn dev
yarn run v1.22.19
$ run-p dev:*
$ cross-env NODE_ENV=development nodemon --require dotenv/config ./server.js --watch ./server.js
$ NODE_ENV=development remix dev
[nodemon] 2.0.21
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): server.js
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node --require dotenv/config ./server.js`
Express server listening on port 3000
💿 Built in 238ms
GET / 200 - - 41.788 ms
💿 File created: app/routes/about.tsx
💿 Rebuilding...
💿 Rebuilt in 110ms
💿 Waiting for http://192.168.1.111:3000...
GET /__REMIX_ASSETS_MANIFEST 200 - - 1.110 ms
💿 http://192.168.1.111:3000 ready in 7ms
/Users/dylan/dev/hmr-example/node_modules/@remix-run/dev/dist/hmr.js:33
    let file = config.routes[routeId].file;
                                      ^

TypeError: Cannot read properties of undefined (reading 'file')
    at Object.updates (/Users/dylan/dev/hmr-example/node_modules/@remix-run/dev/dist/hmr.js:33:39)
    at onRebuildFinish (/Users/dylan/dev/hmr-example/node_modules/@remix-run/dev/dist/devServer2.js:159:27)

Node.js v18.14.2
error Command failed with exit code 1.
```

### Expected Behavior

Adding/moving routes while HMR is running should work.

### Actual Behavior

Dev server crashes.

**Comment by andrecasal** [comment]
I have auto-saved enabled and while typing out new (yet invalid) code, the dev server regularly crashes.

**Comment by huw** [comment]
I’ll add that this also happens for `.css.ts` files with Vanilla Extract, but it’s an upstream issue and I’ve gotten that merged in vanilla-extract-css/vanilla-extract#1024.

If a contributor is fixing this issue do you mind also updating Vanilla Extract?

**Comment by MichaelDeBoey** [comment]
@huw I know @markdalgleish is working on some vanilla extract things right now, so he can probably also update to latest version in the same PR

**Comment by kentcdodds** [comment]
I'm guessing this is the same issue as #5631

**Comment by dmarkow** [comment]
My exact reproduction is still happening in 1.14.3... I don't think this is the same as the "syntax error exists server" issue (which I assume 1.14.3 fixed).

```
💿 Built in 239ms
💿 File created: app/routes/about.tsx
💿 Rebuilding...
💿 Rebuilt in 104ms
💿 Waiting for http://192.168.1.111:3000...
GET /__REMIX_ASSETS_MANIFEST 200 - - 17.550 ms
💿 http://192.168.1.111:3000 ready in 27ms
/Users/dylan/dev/test-app/node_modules/@remix-run/dev/dist/hmr.js:33
    let file = config.routes[routeId].file;
                                      ^

TypeError: Cannot read properties of undefined (reading 'file')
    at Object.updates (/Users/dylan/dev/test-app/node_modules/@remix-run/dev/dist/hmr.js:33:39)
    at onRebuildFinish (/Users/dylan/dev/test-app/node_modules/@remix-run/dev/dist/devServer2.js:159:27)

Node.js v18.14.2
error Command failed with exit code 1.
```

```
> cat package.json | grep remix
    "build": "remix build",
    "dev:remix": "remix dev",
    "@remix-run/express": "^1.14.3",
    "@remix-run/node": "^1.14.3",
    "@remix-run/react": "^1.14.3",
    "@remix-run/dev": "^1.14.3",
    "@remix-run/eslint-config": "^1.14.3",
```

**Comment by pcattori** [comment]
Fixed by #6165
